src/main/java/io/stargate/sgv2/jsonapi/service/schema/naming/NamingRule.java

```diff
@@ -0,0 +1,39 @@
+package io.stargate.sgv2.jsonapi.service.schema.naming;
+
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+
+/** Define a naming rule that can be applied to a string */
+public abstract class NamingRule {
+  private final String name;
+  private final SchemaObject.SchemaObjectType schemaType;
```

Comment:
I think we need to change this a bit: `name` belongs here (it's the name of thing being validated), but `schemaType` should go up to `SchemaObjectNamingRule`. This because we will have other `NamingRule`s for (Collection) Fields and maybe (Table) Columns.